syft/frameworks/torch/tensors/interpreters/functional_encrypted.py

```diff
@@ -0,0 +1,47 @@
+from syft.generic.tensor import AbstractTensor
+import pyfe as fe
+
+
+class FunctionalEncryptedTensor(AbstractTensor):
+    def __init__(self, owner=None, id=None, tags=None, description=None):
+        """Initialize a FunctionalEncryptedTensor, whose behaviour is to encrypt a
+        tensor, on which functional encryption can be used
+
+        Args:
+            owner: An optional BaseWorker object to specify the worker on which
+                the tensor is located.
+            id: An optional string or integer id of the CKKSTensor.
+            tags: an optional set of hashtags corresponding to this tensor
+                which this tensor should be searchable for
+            description: an optional string describing the purpose of the
+                tensor
+         """
+        super().__init__(id=id, owner=owner, tags=tags, description=description)
+        self.scheme = None
+
+    def encrypt(self, scheme):
+        """
+        This method will encrypt each value in the tensor using Functional encryption.
+
+        Args:
+            scheme: a PyFE Scheme created using
+                pyfe.scheme.Scheme()
+        """
+        output = FunctionalEncryptedTensor()
+        output.child = self.child
+        output.encrypt_(scheme)
+
+    def encrypt_(self, scheme):
+        """
+            This method will encrypt each value in the tensor using Functional encryption.
+
+        Args:
+            scheme: a PyFE Scheme created using
+                pyfe.scheme.Scheme()
+        """
+        input = self.child.flatten().tolist()
+        pk, msk = scheme.setup(vector_length=len(input))
```

Comment:
I guess the setup should not be depending of a Tensor. It should be done by a worker [kind of a crypto provider] which then shares the pk to the other workers for encryption. Andd it should never share the msk.

Reply:
It depends just on the one thing though - the length of the vector. But we should have a third party secure worker that generates the pk and msk and share only pk with worker. Will work on that :+1: